### PR TITLE
Accton AS7716_32X: add to support ONIE

### PR DESCRIPTION
**Pull request by david56**
The machine has 32x100GbE QSFP ports on Broadcom Tomahawk switch
ASIC.  CPU module is Intel Xeon D-1518.

### CODE REVIEW

File: machine/accton/accton_as7716_32x/busybox/conf/config [hunk 1]
```diff
@@ -0,0 +1,7 @@
+CONFIG_SYS_EEPROM=y
+CONFIG_SYS_EEPROM_DEVICE_I2C=y
+CONFIG_SYS_EEPROM_I2C_DEVICE="/dev/i2c-0"
+CONFIG_SYS_EEPROM_I2C_ADDR=0x56
+CONFIG_SYS_EEPROM_OFFSET=0
+CONFIG_SYS_EEPROM_SIZE=256
+CONFIG_SYS_EEPROM_MAX_SIZE=256
```
Comment (cbrune) [inline machine/accton/accton_as7716_32x/busybox/conf/config]: Is the EEPROM really this small?

File: machine/accton/accton_as7716_32x/kernel/mgmt-port-init-config.patch [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+diff --git a/drivers/net/ethernet/broadcom/tg3/tg3.c b/drivers/net/ethernet/broadcom/tg3/tg3.c
+index 87a7911..4ba76b8 100644
+--- a/drivers/net/ethernet/broadcom/tg3/tg3.c
++++ b/drivers/net/ethernet/broadcom/tg3/tg3.c
+@@ -1628,6 +1628,7 @@ static void tg3_mdio_config_5785(struct tg3 *tp)
+ static void tg3_mdio_start(struct tg3 *tp)
+ {
+ 	tp->mi_mode &= ~MAC_MI_MODE_AUTO_POLL;
++	tp->mi_mode |= MAC_MI_MODE_SHORT_PREAMBLE;
```
Comment (cbrune) [inline machine/accton/accton_as7716_32x/kernel/mgmt-port-init-config.patch]: Can you elaborate on why this change was necessary?

File: machine/accton/accton_as7716_32x/kernel/mgmt-port-init-config.patch [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+ 	udelay(80);
+ 
+@@ -2899,6 +2900,11 @@ static int tg3_phy_reset(struct tg3 *tp)
+ 		}
+ 	}
+ 
++	if (tp->phy_id == TG3_PHY_ID_BCM5718S) {
++		__tg3_writephy(tp, 0x8, 0x10, 0x1d0);
++		__tg3_writephy(tp, 0x1f, 0x4, 0x5e1);
++	}
```
Comment (cbrune) [inline machine/accton/accton_as7716_32x/kernel/mgmt-port-init-config.patch]: Same thing here.  Can you elaborate on why this change was necessary?
Is it possible to #define a few constants and make this more readable?  It is unclear what this is doing.

File: machine/accton/accton_as7716_32x/kernel/series [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# This series applies on GIT commit 0bffbdd6d4f0accfbf1645111726ab11c38dd780
+driver-broadcom-tigon3.patch
```
Comment (cbrune) [inline machine/accton/accton_as7716_32x/kernel/series]: Where is this driver-broadcom-tigon3.patch?  I do not see it in the review.

**Comment by cbrune** [comment]
The driver-broadcom-tigon3.patch is huge.  Is this a back port of an upstream patch?  If it is a straight back port (no Accton specifics), it would be better has a general 3.2 kernel patch and not platform specific.

Another thing to consider -- I see you are basing the new board on the older 3.2.y kernel series.  Would the tigon3 patch even be necessary if you used the 4.1.y kernel series?

File: machine/accton/accton_as7716_32x/busybox/conf/config (hunk 1 above)
Comment (david56) [inline machine/accton/accton_as7716_32x/busybox/conf/config]: Yes, it is a 2Kbits EEPROM.

File: machine/accton/accton_as7716_32x/kernel/mgmt-port-init-config.patch (hunk 3 above)
Comment (david56) [inline machine/accton/accton_as7716_32x/kernel/mgmt-port-init-config.patch]: The management port MAC which has internal PHY (0x8) connects to external PHY (0x1f).  ``mgmt-port-init-config.patch`` has the following initializations:

1. set internal PHY to SGMII mode that can connect to external PHY
2. set MAC_MI_MODE_SHORT_PREAMBLE for accessing external PHY
3. set basic auto-negotiation capabilities for external PHY

**Comment by david56** [comment]
No, it is not a back port from upstream.  It is a standalone driver came from Broadcom.  For the management port MAC, the driver should be necessary for every kernel version.

**Comment by cbrune** [comment]
Looks great.  Pushed to master:

- 9fd2da8 Accton AS7716_32X: add to support ONIE
